kornia/losses/focal.py

```diff
@@ -68,19 +75,30 @@ def focal_loss(
         f"input and target must be in the same device. Got: {input.device} and {target.device}",
     )
 
-    # compute softmax over the classes axis
-    input_soft: Tensor = input.softmax(1)
-    log_input_soft: Tensor = input.log_softmax(1)
-
     # create the labels one hot tensor
     target_one_hot: Tensor = one_hot(target, num_classes=input.shape[1], device=input.device, dtype=input.dtype)
 
-    # compute the actual focal loss
-    weight = torch.pow(-input_soft + 1.0, gamma)
-
-    focal = -alpha * weight * log_input_soft
-    loss_tmp = torch.einsum('bc...,bc...->b...', (target_one_hot, focal))
+    # compute softmax over the classes axis
+    log_input_soft: Tensor = input.log_softmax(1)
 
+    # compute the actual focal loss
+    loss_tmp = -torch.pow(1.0 - log_input_soft.exp(), gamma) * log_input_soft * target_one_hot
+    if alpha is not None:
+        num_of_classes = input.shape[1]
+        if isinstance(alpha, float):
+            alpha_fac = torch.tensor([1 - alpha] + [alpha] * (num_of_classes - 1))
```

Comment:
```suggestion
            alpha_fac = tensor([1 - alpha] + [alpha] * (num_of_classes - 1))
```
`from kornia.core import tensor`